data/lib/core/achievements.lua

```diff
@@ -650,7 +650,7 @@ end
 
 function isAchievementSecret(ach)
 	local achievement
-	if tonumber(ach) ~= nil then
```

Comment:
Are you sure this is similar? Wouldn't a return value of 0 also be falsy in this case?

Reply:
No, with [testes](http://tpcg.io/0YHQAG), 0 != nil in this case

Reply:
Uh, very odd, Lua is the only language I know that does not consider 0 to be falsy...